lib/virsh_autotest_base.pm

```diff
@@ -0,0 +1,126 @@
+# VIRSH TEST MODULE BASE PACKAGE
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+#
+# Summary: This is the base package for virsh test modules, for example,
+# tests/virtualization/xen/hotplugging.pm
+# tests/virt_autotest/virsh_internal_snapshot.pm
+# tests/virt_autotest/virsh_external_snapshot.pm
+# and etc.
+# Maintainer: Wayne Chen <wchen@suse.com>
+
+package virsh_autotest_base;
```

Comment:
This pkg name is a little bit misleading, especially for those not familiar with virtualization code structure. Can you give it another name?

Reply:
I think this name can represent the test content. If you have other suggestions, please name it.

Reply:
What about virt_feature_test_base? @pdostal Any better suggestions for this base class to better indicate class hierarchy?

Reply:
IMHO, this is not very convincing. This module does not represent any directory within its hierarchy. It represents the actual test contents.

Reply:
> I would say `virtualization_base.pm` as it match the `tests/virtualization/` directory.

@pdostal This file lib/virsh_autotest_base.pm is going to be the base class for recently implemented and future implemented virtualization feature tests, which wants to use such junit log for job review when test done. I have to say these files are under different direcotries. Some of them are in tests/virt_autotest/virsh_internal_snapshot.pm, some of then are in tests/virtualization/xen/hotplugging.pm. So using directory for naming is not a good enough way, i am afraid.

Reply:
@alice-suse than your suggestion if probably the best one :)

Reply:
@pdostal Not all test modules come from tests/virtualization. And this new base module file should not have a so general name, it should reflect what is going to be built upon it.

Reply:
@waynechen55 Then both pavel and me would suggest to use file name virt_feature_test_base for this one. If you agree too, please make the change.

Reply:
Actually all test contents belonging to "Managing virtual machines using libvirt" that uses virsh command which is at the center of topic. But in light of this may also involves feature tests, so I agree upon using virt_feature_test_base. Done.